src/main/java/appeng/me/storage/NetworkInventoryHandler.java

```diff
@@ -34,41 +30,54 @@
 import appeng.api.storage.data.IAEStack;
 import appeng.api.storage.data.IItemList;
 import appeng.me.cache.SecurityCache;
-import appeng.util.ItemSorters;
+import appeng.util.SortedArrayList;
 
 public class NetworkInventoryHandler<T extends IAEStack<T>> implements IMEInventoryHandler<T> {
 
     private static final ThreadLocal<LinkedList> DEPTH_MOD = new ThreadLocal<>();
     private static final ThreadLocal<LinkedList> DEPTH_SIM = new ThreadLocal<>();
-    private static final Comparator<Integer> PRIORITY_SORTER = (o1, o2) -> ItemSorters.compareInt(o2, o1);
+
+    /**
+     * Sorter for the {@link #priorityInventory} list. Sticky inventories are always first. The inventories are then
+     * sorted by priority (highest first), and then by placement pass (1 first, 1&2 second, 2 last).
+     */
+    private static final Comparator<IMEInventoryHandler<?>> STICKY_PRIORITY_PLACEMENT_PASS_SORTER = (o1, o2) -> {
+        int result = Boolean.compare(o2.getSticky(), o1.getSticky());
```

Comment:
I see the problem here:
Item is not counted in craft if it configured to be stored on `Storage Cell` with sticky card installed also it affects craft count into partitioned storage cells. 
This happened because crafting monitors included as pass-thru storages with maximal priority `Integer.MAX_VALUE`, I propose to slightly change logic here and count sticky inventories as storages with priority `Integer.MAX_VALUE  - 1` with additional changes in underlying algorithm

Reply:
I think such a behavior change should be implemented in a different PR so people can see and discuss it. That old code prioritizes sticky inventories over everything as well.

Reply:
Just partitioned storage cells also prevent ability to count crafted items on it and sticky inventories added no so much time before it. It looks like a bug and should be fixed.

Reply:
I think it would be best to ignore it for now and make a separate pr to fix the priority for sticky inventories. This pr aims to make performance improvements but keeping the same functionality so it should not be mutated to also change existing functionality.

Reply:
Ok, let's wind out it into separate PR if @tth05 can take it, but lets describe current behaviour to be sure that we understand correctly what should be done after.

Reply:
Everything with sticky cards takes maximum priority because there's special code that handles them first. Everything else uses the normal priority system, therefore `CraftingGridCache` will come first.